### PR TITLE
Support using MSVC intrinsics in Log2FloorNonZero

### PR DESCRIPTION
**Pull request by KindDragon**

### CODE REVIEW

File: src/google/protobuf/stubs/port.h [hunk 1]
```diff
@@ -394,12 +395,10 @@ class Bits {
...
   return 31 ^ static_cast<uint32>(__builtin_clz(n));
-#elif defined(COMPILER_MSVC) && defined(_M_IX86)
-  _asm {
-    bsr ebx, n
-    mov n, ebx
-  }
-  return n;
+#elif defined(_MSC_VER)
+  unsigned long where;
+  _BitScanReverse(&where, n);
```
Comment (xfxyjwf) [inline src/google/protobuf/stubs/port.h]: Is this more efficient than the original assembly?

Comment (KindDragon) [reply src/google/protobuf/stubs/port.h]: No, it's just as effective but works also for 64bit build (inline assembler can only be used in 32-bit compiler `_M_IX86`)

**Comment by KindDragon** [comment]
Hi, @xfxyjwf. Can you review this merge request?

**Comment by KindDragon** [comment]
Thanks

**Pull request merged**